### ISSUE TITLE
Improved PotentialBarrier Handling

### ISSUE THREAD

**Issue by boldtrn** [graphhopper/graphhopper#1161]
Currently, we check if a potential barrier is an actual barrier like this:

[OSMReader](https://github.com/graphhopper/graphhopper/blob/91ca93d8520480f3d103a79962ecabad35f72c78/reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java#L526):
```
long nodeFlags = encodingManager.handleNodeTags(node);
if (nodeFlags != 0)
   getNodeFlagsMap().put(node.getId(), nodeFlags);
``` 

The `nodeFlags` either contain the information `barrier=true|false` for multiple encoders. When we call this method, we haven't read the ways yet, so we cannot create a correlation with ways yet.

Later on, [we have access to the way](https://github.com/graphhopper/graphhopper/blob/91ca93d8520480f3d103a79962ecabad35f72c78/reader-osm/src/main/java/com/graphhopper/reader/osm/OSMReader.java#L369), but don't have access to the node anymore, we only have the information if the node was a barrier or not. If the node was a barrier, we split the way and make it impossible for the corresponding vehicle to cross the way.

This process works perfect for absoluteBarriers, e.g., a car won't be able drive over a bollard. However there are other barriers that might work different. The classic example would be a gate. AFAIK there exists no real consensus on what the implied access exists for gate. Especially outside of Central Europe, the OSM data quality is often "improvable". But even in Central Europe the access for gates is not tagged. Fixing the access tag for every barrier, might be nice, but is unrealistic.

If you look around in OSM, it seems that the access of the gate should be derived from the access type of the ways that are on either side of the barrier. For example look at this [overpass query](http://overpass-turbo.eu/s/rWy), which shows barriers that do not contain any access information but are mapped on major roads. In Germany I haven't found many gates which were not tagged. But if you look at different countries, there are a lot more examples where major roads might be blocked due to that.

I think it is quite safe to assume that gates that are placed on a highway=primary|secondary|tertiary are accessible. Right now there is no way for us to check this. 

I wrote a lot of notes on OSM so that maybe the barriers will be clearly mapped. Still I think it would be a nice addition to GraphHopper to include these kind of checks. WDYT?

**Comment by karussell** [comment]
Yes, a simple possibility would be to create an edge that does not block but has some access values associated so that it can be dynamically changed. I.e. the flags that are returned have not just access encoded values. But let us do #1112 first.
